sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -472,10 +423,13 @@ export class MessageSender extends LinkEntity {
    */
   async onDetached(senderError?: AmqpError | Error): Promise<void> {
     try {
-      const wasCloseInitiated = this._sender && this._sender.isItselfClosed();
+      const wasCloseInitiated = this.link && this.link.isItselfClosed();
       // Clears the token renewal timer. Closes the link and its session if they are open.
       // Removes the link and its session if they are present in rhea's cache.
-      await this._closeLink(this._sender);
+      if (this.link) {
```

Comment:
Nit: Why the check on link here where as in other places we call `this.closeLink()` without this check assuming that we do the relevant checks inside `closeLink()`

Reply:
Removed. I also added in a test just to ensure that it's safe to call closeLink multiple times. (it already was, but the tests are nice).